rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -0,0 +1,1424 @@
+            model_dir / f"{file_name}.inverted_label_dict.pkl"
+        )
+        inverted_tag_dict = io_utils.json_unpickle(
+            model_dir / f"{file_name}.inverted_tag_dict.pkl"
+        )
+        batch_tuple_sizes = io_utils.pickle_load(
+            model_dir / f"{file_name}.batch_tuple_sizes.pkl"
+        )
+
+        # jsonpickle converts dictionary keys to strings
+        inverted_label_dict = {
+            int(key): value for key, value in inverted_label_dict.items()
+        }
+        if inverted_tag_dict is not None:
+            inverted_tag_dict = {
+                int(key): value for key, value in inverted_tag_dict.items()
+            }
+
+        return (
+            batch_tuple_sizes,
+            inverted_label_dict,
+            inverted_tag_dict,
+            label_data,
+            meta,
+            data_example,
+        )
+
+    @classmethod
+    def _load_model(
+        cls,
+        inv_tag_dict: Dict[int, Text],
+        label_data: RasaModelData,
+        meta: Dict[Text, Any],
+        data_example: Dict[Text, List[np.ndarray]],
+        model_dir: Text,
+    ):
+        file_name = meta.get("file")
+        tf_model_file = os.path.join(model_dir, file_name + ".tf_model")
+
+        label_key = "label_ids" if meta[INTENT_CLASSIFICATION] else None
+        model_data_example = RasaModelData(label_key=label_key, data=data_example)
+
+        model = cls.model_class().load(
+            tf_model_file,
+            model_data_example,
+            data_signature=model_data_example.get_signature(),
+            label_data=label_data,
+            inverted_tag_dict=inv_tag_dict,
+            config=meta,
+        )
+
+        # build the graph for prediction
+        predict_data_example = RasaModelData(
+            label_key=label_key,
+            data={k: vs for k, vs in model_data_example.items() if "text" in k},
+        )
+
+        model.build_for_predict(predict_data_example)
+
+        return model
+
+
+# accessing _tf_layers with any key results in key-error, disable it
+# pytype: disable=key-error
+
+
+class DIET(RasaModel):
+    def __init__(
+        self,
+        data_signature: Dict[Text, List[FeatureSignature]],
+        label_data: RasaModelData,
+        inverted_tag_dict: Optional[Dict[int, Text]],
+        config: Dict[Text, Any],
+    ) -> None:
+        super().__init__(name="DIET", random_seed=config[RANDOM_SEED])
+
+        self.config = config
+
+        self.data_signature = data_signature
+        self._check_data()
+
+        self.predict_data_signature = {
+            k: vs for k, vs in data_signature.items() if "text" in k
+        }
+
+        label_batch = label_data.prepare_batch()
+        self.tf_label_data = self.batch_to_model_data_format(
+            label_batch, label_data.get_signature()
+        )
+        self._num_tags = len(inverted_tag_dict) if inverted_tag_dict is not None else 0
+
+        # tf objects
+        self._tf_layers = {}
```

Comment:
let's add type annotations here

Reply:
what do you mean? for this dict?

Reply:
yes, would be great to know what's in there